client/inbox-panel/index.js

```diff
@@ -85,6 +170,103 @@ const InboxPanel = ( props ) => {
 		updateUserPreferences( userDataFields );
 	}, [] );
 
+	const onDismiss = ( note, type ) => {
+		setDismiss( { note, type } );
+	};
+
+	const closeDismissModal = ( confirmed = false ) => {
+		const noteNameDismissAll = dismiss.type === 'all' ? true : false;
+		const screen = getScreenName();
+
+		recordEvent( 'inbox_action_dismiss', {
+			note_name: dismiss.note.name,
+			note_title: dismiss.note.title,
+			note_name_dismiss_all: noteNameDismissAll,
+			note_name_dismiss_confirmation: confirmed,
+			screen,
+		} );
+
+		if ( confirmed ) {
+			const noteId = dismiss.note.id;
+			if ( noteId && ! noteNameDismissAll ) {
+				removeNote( noteId )
```

Comment:
Yeesh this indentation is... blech..

Could we use async/await here?

Reply:
I refactored it a bit here: https://github.com/woocommerce/woocommerce-admin/pull/7006/commits/49820bb5a009bd74ccbd8a26e2d9f18c56d8d47b I did use async/away, which only really took away one indentation, I streamlined it a bit more, so it is less lengthy.